pyfar/dsp/fractional_octave_smoothing.py

```diff
@@ -0,0 +1,317 @@
+        upper_limit.reshape(1, -1),
+        lower_limit.reshape(1, -1)))
+
+
+def data_padder(data, pad_width, mean_size):
+    """
+    Pads data array of shape (N, M) to data matrix of shape
+    (N, M, M+pad_width). The output data contains M copies of the input data
+    rows padded by 'pad_width' with the mean value of the last data samples.
+    The number of samples used to compute the mean for each row is specified in
+    the array 'mean_size'.
+
+    Parameters
+    ----------
+    data : ndarray
+        Array of shape (N, M) of the data.
+        (N= Number of channels, M=Length of Signal)
+    pad_width : int
+        Padding length
+    mean_size : ndarray
+        Array with numbers of samples used to compute the mean.
+
+    Returns
+    -------
+    ndarray
+        Padded data matrix containing the data values and specified padding
+        value for each frequency bin.
+
+    Raises
+    ------
+    ValueError
+        Data array must be 2D.
+    """
+    if not data.ndim == 2:
+        raise ValueError('Data array must be 2D.')
+    padded = np.array([np.pad(data, ((0, 0), (0, pad_width)),
+                              'mean', stat_length=(m)) for m in mean_size])
+    # move channel axis to front and return
+    return np.moveaxis(padded, 1, 0)
+
+
+def polar2cartesian(amplitude, phase):
+    """
+    Converts two arrays of amplitude and phase into one array
+    of complex numbers in cartesian form.
+
+    Parameters
+    ----------
+    amplitude : ndarray
+        Array of amplitude values.
+    phase : ndarray
+        Array of phase values.
+
+    Returns
+    -------
+    ndarray
+        Array of same shape as input arrays with complex numbers.
+
+    Raises
+    ------
+    ValueError
+        Arrays must have same shape.
+    """
+    if not (amplitude.shape == phase.shape):
+        raise ValueError("Arrays must have same shapes.")
+    # Save input shape
+    input_shape = amplitude.shape
+    # Reshape for cmath rect to 1D arrays:
+    reshaped_amplitude = amplitude.reshape(1, -1)[0]
+    reshaped_phase = phase.reshape(1, -1)[0]
+    # Apply cmath rect on elements:
+    cartesian = np.array([cm.rect(a, p) for a, p in
+                          zip(reshaped_amplitude, reshaped_phase)])
+    return cartesian.reshape(input_shape)
+
+
+def frac_smooth_signal(signal, smoothing_width):
+    """
+    Method to smooth a given signal.
+    Takes the data of a given signal of shape (n, m), where n is number of
+    channels and m length of the signal.
+    Creates an object of class FractionalSmoothing to compute integration
+    limits and smoothing weights. Finally applies the weights on the input data
+    and returns smoothed signal.
+
+    Parameters
+    ----------
+    signal : Signal
+        Input signal to be smoothed
+    smoothing_width : float, int
+        Width of smoothing window relative to an octave
+
+    Returns
+    -------
+    Signal
+        Smoothed Signal
+
+    Raises
```

Comment:
We usually do not include the `Raises` part. What should be included is a reference to the paper after which this is implemented
```suggestion
    References
    ----------
    .. [1] Name, Title, Journal, ...
```

You can use _[1] inside the docstring to link to the refernece

Reply:
Done.